core/base/scalarFieldCriticalPoints/ScalarFieldCriticalPoints.h

```diff
@@ -508,13 +555,16 @@ template <class triangulationType>
 char ttk::ScalarFieldCriticalPoints::getCriticalType(
   const SimplexId &vertexId,
   const SimplexId *const offsets,
-  const triangulationType *triangulation) const {
+  const triangulationType *triangulation,
+  std::vector<std::vector<ttk::SimplexId>> *upperComponents,
```

Comment:
here set a default value `upperComponents = nullptr` (same for `lowerComponents`)

Reply:
then, in the code (before the call to `getLowerAndUpperComponents`), instantiate a `std::vector<std::vector<ttk::SimplexId>> localUpperComponents;`
then:
```
if(upperComponents == nullptr){
  upperComponents= &localUpperComponents;
}
```

same for `lowerComponents`.